index.js

```diff
@@ -22,19 +22,28 @@ const kLast = Symbol('last')
 const kDecoder = Symbol('decoder')
 
 function transform (chunk, enc, cb) {
-  this[kLast] += this[kDecoder].write(chunk)
-  if (this[kLast].length > this.maxLength) {
-    return cb(new Error('maximum buffer reached'))
+  var list
+  if (this.overflow) {
+    var buf = this[kDecoder].write(chunk)
+    list = buf.split(this.matcher)
+    if (list.length === 1) return cb()
```

Comment:
I don't understand this `return cb()`

Reply:
String#split() returns an array containing one element in exactly one case. If the length of `list` is 1 here, then it means the decoded chunk did not contain the matcher and we therefore want to ignore this chunk (because it followed an overflow) by returning without any push or buffering. 

If we continued to buffer while overflowed, then we pay a substantial cost in terms of memory and time, which would defeat the purpose of adding this new option.

Reply:
Can you put a comment explaining all of this?